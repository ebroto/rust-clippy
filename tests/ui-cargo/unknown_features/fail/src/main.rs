// compile-flags: --crate-name=unknown_features --cfg feature="misspelled" --cfg feature="another"
#![warn(clippy::unknown_features)]

fn main() {
    #[cfg(feature = "mispelled")]
    let _ = 42;

    #[cfg(feature = "dependency/unknown")]
    let _ = 42;

    #[cfg(any(not(feature = "misspeled"), feature = "not-found"))]
    let _ = 21;

    if cfg!(feature = "nothe") {}
}
