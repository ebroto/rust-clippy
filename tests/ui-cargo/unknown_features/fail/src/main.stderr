error: unknown feature `mispelled`
  --> $DIR/main.rs:5:21
   |
LL |     #[cfg(feature = "mispelled")]
   |                     ^^^^^^^^^^^ help: a feature with a similar name exists: `"misspelled"`
   |
   = note: `-D clippy::unknown-features` implied by `-D warnings`

error: unknown feature `dependency/unknown`
  --> $DIR/main.rs:8:21
   |
LL |     #[cfg(feature = "dependency/unknown")]
   |                     ^^^^^^^^^^^^^^^^^^^^

error: unknown feature `misspeled`
  --> $DIR/main.rs:11:29
   |
LL |     #[cfg(any(not(feature = "misspeled"), feature = "not-found"))]
   |                             ^^^^^^^^^^^ help: a feature with a similar name exists: `"misspelled"`

error: unknown feature `not-found`
  --> $DIR/main.rs:11:53
   |
LL |     #[cfg(any(not(feature = "misspeled"), feature = "not-found"))]
   |                                                     ^^^^^^^^^^^

error: unknown feature `nothe`
  --> $DIR/main.rs:14:23
   |
LL |     if cfg!(feature = "nothe") {}
   |                       ^^^^^^^ help: a feature with a similar name exists: `"another"`

error: aborting due to 5 previous errors

